### ISSUE TITLE
Add support for weighted loss functions

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j#1329]
Essentially: allow a user to provide a weight for learning, with loss functions such as negative log likelihood and MSE.

**Comment by treo** [comment]
Do I understand this correctly, that this would allow to weight my examples / input, such that I can train my model with an unbalanced training set?

**Comment by AlexDBlack** [comment]
I was thinking a weight across the output, but also weighting *examples* is possible too.
For example, http://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_squared_error.html has both weights for each input (sample_weight) and a weight across each output value (multioutput)

**Comment by treo** [comment]
What would one use the weight across the output for? The scikit-learn doc confused me more than it cleared things up.

**Comment by AlexDBlack** [comment]
https://github.com/deeplearning4j/nd4j/pull/1357
